### PR TITLE
Async_redux

### PR DESCRIPTION
**Pull request by HeDevedUp**
## Description

**Task:** https://github.com/microverseinc/curriculum-transversal-skills/blob/main/code-review/exercises/full_partner_review.md

This PR makes the following changes:
-  Sets up the GitHub Actions and the [Stylelint](https://stylelint.io/), and [ESLint](https://eslint.org/) linters
-  Sets up the Webpack configuration files and builds scripts

## Type of change
- [X]  used reduxAsyncThunk to fetch from the api
- [X]  added redux to my component using useDispatch() and useSelector
- [X] npm install  redux with  @reduxjs/toolkit
- [X] I created a redux directory 
- [X] created a store 
- [X] created a component for book and input form 
- [X] passed props within component 
- [X] received props from component
- [X] created input froms
- [X] added an input
- [X] created a react app with  npx create-react-app my-app
- [X] added linters for react module 
- [X]  added boilerplate for the calculator 
- [X]  used CSS to design the boilerplate 
- [X] refactored the code base and added a component folder
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented on my code
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: src/components/book/bookContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from 'react'
```
Comment (Ismailco) [inline src/components/book/bookContainer.jsx:1]: - [ ] The App produces errors, kindly fix them so your app is working as intended 

Screenshot_20220714_175001

File: src/redux/book/book.js [hunk 2]
```diff
@@ -1,41 +1,57 @@
...
-    author: 'author 3',
-    id: 3,
-  },
-];
-
-// reducer
-
-export default (state = initialState, action) => {
+import { getData, removeData, uploadData } from '../../util/axios';
+// Actions
```
Comment (Ismailco) [inline src/redux/book/book.js]: - [ ] `createAsyncThunk` should be used in the project for creating thunk action creators, kindly implement this it's a requirement

File: src/redux/book/book.js [hunk 3]
```diff
@@ -1,41 +1,57 @@
...
+    }
+    default:
+      return initialState;
   }
+}
+
+// Action creators
+
+// export const addbook = (book) => ({ type: ADDBOOK, payload: book });
+export const addbook = (book) => async (dispatch) => {
```
Comment (Ismailco) [inline src/redux/book/book.js]: - [ ] Thunk action creator for fetching books should be created, kindly implement this it's a requirement

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @HeDevedUp, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @HeDevedUp, 

Thank you for implementing changes requested by previous reviewer. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
## Highlights 
- Clean and well organized project✔️
- Professional readme file ✔️

# Status: Approved ✔️✔️✔️

## Optional
For your next project remember to make all your commit messages capitalized and imperative.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**